features/FEATURE_BLE/source/generic/GenericGap.cpp

```diff
@@ -1084,7 +1231,17 @@ void GenericGap::on_connection_complete(const pal::GapConnectionCompleteEvent& e
             address.data(),
             &connection_params
         );
-    } else {
+
+        // Now starts pairing or authentication procedures if required
+        if(needs_pairing) {
+            SecurityManager &sm = createBLEInstance()->getSecurityManager();
+            sm.requestPairing(e.connection_handle);
```

Comment:
requestPairing is for the central role; you must use request authentication here.